Automattic/jetpack
# Jetpack Manage: Add a Jetpack Manage banner to My Jetpack

Fixes https://github.com/Automattic/jetpack-manage/issues/154

## Proposed changes:

This PR adds a Jetpack Manage banner to My Jetpack. The site requires to be connected, and the user should have two or more sites. 

If the user is not yet a partner/agency, it will show the "Sign up for free" button, and it will link to `https://cloud.jetpack.com/agency/signup/` using the redirection `my-jetpack-jetpack-manage-sign-up`:

<img width="1090" alt="image" src="https://github.com/Automattic/jetpack/assets/9832440/625436f7-ee2b-4c85-8f47-655b6b641d49">

If the user is already a partner/agency, it will show the "Manage sites" button, and it will link to `https://cloud.jetpack.com/dashboard/` using the redirection `my-jetpack-jetpack-manage-dashboard`
<img width="1101" alt="image" src="https://github.com/Automattic/jetpack/assets/9832440/b2deb563-c9eb-49ef-876f-76d1940c11dd">

I created a new component, `UpsellBanner`, to be reused. It is used by `JetpackManageBanner` component also created in this PR. For now, it's inside My Jetpack package, but it could be extracted into `@automattic/jetpack-components`. 

**Pending tasks**

- [x] Create the Jetpack redirection: `my-jetpack-jetpack-manage-dashboard` => `https://cloud.jetpack.com/dashboard/` (confirmed and created)

### Other information:


## Jetpack product discussion

pf36In-s1-p2

## Does this pull request change what data or activity we track or use?



## Testing instructions:

- Check the code
- The Jetpack Manage banner should appear in My Jetpack if the site is connected and the user has two or more connected sites.
- The “Learn more” button should lead to [https://jetpack.com/manage](https://href.li/?https://jetpack.com/manage)
- If the user is not a partner/agency, the second button will be "Sign up for free" and if you click on it, it will be redirected to `https://cloud.jetpack.com/agency/signup/`
   - You can change it in MC `/jetpack-partner-portal/partners` by switching your account to another type, different to Agency. (Note: The account info is cached by 1 hour, maybe you have to launch another JN site to check different cases)
- If the user is a partner/agency, the second button will be "Manage sites" and if you click on it, it will be redirected to `https://cloud.jetpack.com/dashboard/`
